modal/_container_entrypoint.py

```diff
@@ -593,12 +593,14 @@ def breakpoint_wrapper():
     # The only caveat is a bunch of calls will now cross threads, which adds a bit of overhead?
     client = Client.from_env()
 
+    exited_successfully = True
     try:
         with proxy_tunnel(container_args.proxy_info):
             try:
                 main(container_args, client)
             except UserException:
                 logger.info("User exception caught, exiting")
+                exited_successfully = False
```

Comment:
Maybe slightly cleaner in an `else:` block?

Reply:
why not just re-raise

Reply:
that way you don't need any of the other changes, including the `exited_successfully` variable and the `exit(1)` call